fetch_mail.py

```diff
@@ -0,0 +1,160 @@
+"""
+Method1
+-------
+    1. imaplib module
+        -> Establish a connection
+        -> Select the mailbox
+        -> Search for unseen mails
+        -> fetch their id
+        -> get raw data of each id
+        -> perform regex to find msg. patterns
+        -> save into a file
+        -> quit()
+"""
+
+import imaplib  # For reading messages
+import logging  # For catching logs
+import email
+import re
+import email.policy
+import socket  # Require for imaplib. connection
+
+
+class LPForum:
+    def __init__(self, email, password):
+        self.email_user = email
+        self.email_pass = password
+
+    def __str__(self):
+        return str(self.email_user)
+
+    def login(self):
+        """ Creating a secure connection over SSL socket
+        Return an imap4 object to "mail" variable """
+        try:
+            mail = imaplib.IMAP4_SSL("imap.gmail.com", 993)
+
+            # login to user account
+            mail.login(self.email_user, self.email_pass)
+            return mail
+        except imaplib.IMAP4.error as err:
+            logging.info(err)
+            return False
+
+    def get_message_id(self, obj):
+        """ Method for Return the number of mails
+        specified in the mailbox. """
+
+        obj.select(mailbox="INBOX")  # Select a mailbox first
+        _, msgnums = obj.search(
+            "utf-8", "(Unseen)", "FROM", "linkinpark@discoursemail.com"
+        )  # Search message inside mailbox
+        logging.info(f"Found unseen mail list : {msgnums}")
+        if len(msgnums[0]) > 0:
+            return msgnums
+        return False
+
+    def fetch_raw_message(self, obj, message_id):
+        """ Extracting raw message from message id's.
+        message_id contains id's of email messages
+        by linkinpark@discoursemail. 
+        (RFC822) is a msg. format, defines => header, payload
+        """
+
+        ids = message_id[0].split()
+        data = []
+        for id_ in ids:
+            typ, response = obj.fetch(id_, "(RFC822)")
+            data.append(response[0][1])
+
+        if typ == "OK":
+            # raw_text contains two part -> id[0][0] and message body[0][1]
+            return data
+        else:
+            return False
+
+    def extract_contents(self, raw_data: list):
+        """ Parse the message :
+        header  (recipient information)
+        payload (content) """
+
+        # Get Headers
+        Headers = {}
+        messages_obj = [
+            email.message_from_bytes(raw_, policy=email.policy.default)
+            for raw_ in raw_data
```

Comment:
This looks wrong to me, like an empty loop.

Maybe its a python thing I don't know about

Reply:
This is a list comprehension similar to this normal loop : 
```py
message_obj = []
for raw_ in raw_data:
<tab>message_obj.append(email.message_from_bytes(raw_, policy=email.policy.default))
```
Also, email.message_from_bytes() is used to parse the data because in the previous step (imaplib.fetch()) I told you that it returns a very raw data and it is very hard to even read it. So converting that raw data into an email object will allow us to easily parse the email message.  Once you converted it in an email object then get the payload and obtain a text/plain version of email message.